app/controllers/api/v0/events_controller.rb

```diff
@@ -6,11 +6,11 @@ def create
     render(json: error, status: error.status_code) and return if error
 
     inbound_binding.events.each do |event|
-      # first, schema validate the event.data object
-
       event.data[:user_uuid] = current_user_uuid if current_user_uuid
 
-      KafkaClient.produce(data: event.data, topic: event.topic)
+      avro_encoded_data = KafkaAvroTurf.instance.encode(event.data, schema_name: event.type)
+
+      KafkaClient.produce(data: avro_encoded_data, topic: event.topic)
```

Comment:
Eventually we'll want to shoot for asynchronous production to do them in batches https://github.com/a15k/interactions-api/blob/41dba4ed0360d0247fbfa727422b368332b6c38a/lib/kafka_helpers.rb#L22-L28

Reply:
@jpslav was already using a async producer 

https://github.com/openstax/event-capture-api/blob/665c73c587a5e14b382352b3aec0660bd6c248c7/app/services/kafka_client.rb

this is configured via 
https://github.com/openstax/event-capture-api/blob/665c73c587a5e14b382352b3aec0660bd6c248c7/config/secrets.yml.example#L47

Reply:
don't think the validation/encoding needs to be batched - in fact we wan to error fast if sent bad data.

Reply:
@reedstrm in this case, if an encode is bad on a record, it would raise an exception and fall out of this loop w/ the 500 error back to the caller